peyvandak/Majlesi/django-docker-compose-url-shortener/urlshortener_api/api/views.py

```diff
@@ -0,0 +1,177 @@
+from rest_framework import status
+from django.contrib.auth.models import User
+from django.contrib.auth import authenticate, login
+from django.contrib.auth import logout
+from django.contrib.auth.models import User
+from django.utils import timezone
+
+
+@api_view(['POST'])
+def create_user(request):
+    if request.method == 'POST':
+        try:
+            user = User.objects.create_user(request.data['username'], request.data['email'], request.data['password'])
+            user.save()
+            return JsonResponse({"result": "new account has been created"})
+        except:
+            return JsonResponse({"result": 'Please enter "username" , "password", "email"'})
+
+
+@api_view(['GET'])
+def login_user(request):
+    if request.method == 'GET':
+        if request.user.is_authenticated:
+            return JsonResponse({"result": "You have logined before!"})
+        else:
+            user = authenticate(username=request.data["username"], password=request.data["password"])
+            if user is not None:
+                login(request, user)
+                result = "login successfully " + str(request.user.username) + "::"
+                return JsonResponse({"result": result}, safe=False)
+            else:
+                return JsonResponse({"result": "username or password is wrong. Try one more time."}, safe=False)
+
+
+@api_view(['GET'])
+def login_state(request):
+    if request.method == 'GET':
+        if request.user.is_authenticated:
+            logins = request.user.username + " has logined"
+            return JsonResponse({"result": logins}, safe=False)
+        else:
+            return JsonResponse({"result": "No One"}, safe=False)
+
+
+@api_view(['GET'])
+def specific_url_detail(request, shortened_part):
+    if request.method == 'GET':
+        times_followed = 0
+        unique_times_followed = 0
+        long_url = ""
+        try:
+            shortener = Shortener.objects.get(short_url=shortened_part)
+            times_followed = shortener.times_followed
+            unique_times_followed = shortener.unique_times_followed
+            long_url = shortener.long_url
+        except:
+            return JsonResponse({"result": "There isn't any long url with this shortened part"})
+        try:
+            visit_url = list(VisitUrl.objects.filter(short_url=shortened_part).values("ip_address", "visit_date", "device", "response_date", "browser", "browser_version", "os", "os_version", "visitor"))
+            return JsonResponse({
+                "times_followed": times_followed, 
+                "unique_times_followed": unique_times_followed,
+                "long_url": long_url,
+                "visit_data": visit_url
+                }, safe=False)
+        except:
+            return JsonResponse({"result": "No one has visited the link yet"}, safe=False)
+
+
+def logout_user(request):
+    logout(request)
+    return JsonResponse({"result": "logout successfully"}, safe=False)
+
+
+def device(request):
```

Comment:
you can probably move this function to another file for example a `utils.py` file